packages/app/src/systems/Asset/components/AssetInput/AmountInput.tsx

```diff
@@ -0,0 +1,111 @@
+import { cssObj } from '@fuel-ui/css';
+import { Button, Flex, Input, Text } from '@fuel-ui/react';
+import { DECIMAL_UNITS, bn } from 'fuels';
+import type { BN } from 'fuels';
+import { useState } from 'react';
+import type { FC, ChangeEvent } from 'react';
+
+import { AmountInputLoader } from './AmountInputLoader';
+
+export type AmountInputProps = {
+  balance: BN;
+  value: BN;
+  onChange: (val: BN) => void;
+};
+
+type AmountInputComponent = FC<AmountInputProps> & {
+  Loader: typeof AmountInputLoader;
+};
+
+export const AmountInput: AmountInputComponent = ({
+  balance,
+  value,
+  onChange,
+}) => {
+  const [assetAmount, setAssetAmount] = useState<string>(
+    value.eq(0) ? '' : value.toString()
+  );
+
+  const handleAmountChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const newAssetAmount = handleAmountLeadingZeros(event);
+    onChange(bn.parseUnits(event.target.value, DECIMAL_UNITS));
+    setAssetAmount(newAssetAmount);
+  };
+
+  const handlePress = () => {
+    onChange(balance);
+    setAssetAmount(balance.format({ precision: DECIMAL_UNITS }));
+  };
```

Comment:
small logic change - comment 1/2

this 2 methods should use the same logic, having them separated opens space to bugs because someone can forget to update one of the sides.

I will suggest a commit but I didn't test it, it's just for you to understand the idea and use it if you thing makes sense

```suggestion
  const handleAmountChange = (amount: string) => {
    const newAssetAmount = handleAmountLeadingZeros(event);
    onChange(bn.parseUnits(newAssetAmount));
    setAssetAmount(newAssetAmount);
  };

  const handlePress = () => {
    handleAmountChange(balance.format({ precision: DECIMAL_UNITS }));
  };
```

Reply:
That makes sense thanks!